src/App.tsx

```
@@ -1,19 +1,50 @@
-import { About } from './components/About/About.tsx';
-import { Footer } from './components/Footer/Footer.tsx';
-import { Header } from './components/Header/Header.tsx';
+import type { FC, MouseEvent } from 'react';
+import { useState } from 'react';
 
-import styles from './app.module.css';
+import { About } from '@components/About/About';
+import { Footer } from '@components/Footer/Footer';
+import { Header } from '@components/Header/Header';
+import { ProductsList } from '@components/ProductsList/ProductsList';
+
+import { PRODUCTS } from '@/data/mock-products';
+
+import styles from './App.module.css';
+
+const App: FC = () => {
+    const [pageActive, setPageActive] = useState<string>('about');
+
+    const onPageLinkClick = (event: MouseEvent<HTMLAnchorElement>) => {
+        event.preventDefault();
+        const clickedPageLink = event.currentTarget.dataset.page;
+
+        if (clickedPageLink) {
+            setPageActive(clickedPageLink);
+        }
+    };
+
+    let content;
+
+    switch (pageActive) {
+        case 'about': {
+            content = <About />;
+            break;
+        }
+        case 'products': {
+            content = <ProductsList products={PRODUCTS} />;
+            break;
+        }
+        default: {
+            content = <About />;
```

and here, for default you can throw an error